curl/src/client.cc

```diff
@@ -181,17 +181,19 @@ namespace leatherman { namespace curl {
         char errbuf[CURL_ERROR_SIZE] = { '\0' };
         FILE* fp = nullptr;
         fs::path temp_path = fs::path("");
+        bool file_operation_error = false;
         try {
             temp_path = fs::path(file_path).parent_path() / fs::unique_path("temp_file_%%%%-%%%%-%%%%-%%%%");
             fp = boost::nowide::fopen(temp_path.string().c_str(), "wb");
```

Comment:
The logic around closing and cleaning up resources is getting messy to reason about. I'd suggest looking at scoped_resource; maybe use it to close the file and optionally remove it if it exists.

I want to be able to throw the exception I care about, and let RAII handle cleanup.